core/src/main/java/org/apache/carbondata/core/metadata/schema/PartitionInfo.java

```diff
@@ -65,6 +65,31 @@ public PartitionInfo(List<ColumnSchema> columnSchemaList, PartitionType partitio
     this.partitionIds = new ArrayList<>();
   }
 
+  /**
+   * add partition means split default partition, add in last directly
```

Comment:
because maybe there is data existed in default partition need to be filled in new partition